graph/step.go

```diff
@@ -134,6 +146,39 @@ func (s *Step) Validate() error {
 	return nil
 }
 
+// ValidateMounts checks each mount is well formed and each container file path is unique
+func ValidateMounts(mounts []*volume.Mount) error {
+	duplicate := make(map[string]struct{}, len(mounts))
+	for _, m := range mounts {
+		// call m.Validate() for each mount
+		if err := m.Validate(); err != nil {
+			return err
+		}
+		// make sure each container file path provided is unique
+		if _, exists := duplicate[m.MountPath]; exists {
+			return errors.New("mount with duplicate container file path found")
+		}
```

Comment:
is it ok to have two mount path like the following?

/home/user
/home/user/john

Reply:
Docker daemon allows for this since it considers it 2 separate directories. This is the same case as if the customer tried to mount the volume to the /workspace folder. You can't mount to /workspace directly since that's the home volume. But you can mount to a subdirectory like /workspace/sub

Reply:
Upon further investigation, this scenario causes issues when we mount volumes as readonly